### DISCUSSION TITLE
Unable to run Make commands in Bash terminal

### DISCUSSION THREAD

**Discussion by Shane0xM** [Cyfrin/foundry-full-course-cu#1919] [General]
Hello,

I'm currently trying to set up a Chainlink node following the official documentation. I've successfully installed Yarn and cloned the Chainlink repository, but I'm encountering issues when trying to run `make` commands as part of the setup process.

### Environment:
- **OS**: Windows 10
- **Yarn Version**: 1.22.22
- **Chainlink Repository**:  `https://github.com/smartcontractkit/chainlink#install`

### Steps Taken:
1. Installed Yarn and verified the installation.
2. Cloned the Chainlink repository from GitHub to my local machine.
3. Installed MinGW Installation Manager
4. Adjusted PATH in environmental variables
5. Attempted to run `make` commands as specified in the setup instructions.

### Issue Encountered:
When I try to run `make` commands, such as `make install` or `make build`, I receive the following errors: 
1.  bash: mingw32-make: command not found
2.  bash: make: command not found

I've verified that `mingw32-make` is installed and added to my system's PATH.  Yet running `mingw32-make --version` still doesn't work.  I've also tried creating a batch file to alias `mingw32-make` as `make`, but the issue persists.

I'm wondering if there are specific considerations or additional setup steps required for running these `make` commands on Windows, especially for setting up a Chainlink node. Any insights, suggestions, or guidance from the community would be greatly appreciated.

Thank you in advance for your help!

**Comment by Shane0xM** [comment]
This is what I see in my MinGW installation manager:

![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/169729951/02d27eaa-c86e-490b-8961-fbd268724c30)


Based on this I would think the make.exe in my file explorer, but I unfortunately do not. Any ideas on how to best troubleshoot this would be greatly appreciated.

**Reply by EngrPips** [reply]
Wow, it's Great that you are trying to set up a chainlink node. I must confess I have never done this before and don't know how to help, but let's look forward to help from @PatrickAlphaC whenever they are available.

**Reply by Shane0xM** [reply]
Thanks for quick response and for looping @PatrickAlphaC  in here, I actually was working through his tutorial from last year on Chainlink channel to get this set up. 

Ended up spinning my wheels trying to get the `make` commands to work and initialize set up, so I figured I would turn to the community for help, as there sadly aren't many updated resources for set up from source.

**Reply by EngrPips** [reply]
Yeah, setting up stuff always is the most challenging job as far as programming is concerned. Plus, when there are not enough resources to help, it can be frustrating. Let's look forward to Patrick's response.

**Reply by PatrickAlphaC** [reply]
hmm... Thanks for this issue! An AI would likely be really helpful in triaging this. But in the meantime you can just run the target instead of make:

for example, instead of :

```
make install
```

Look in the `Makefile` and whatever it's "pointing to" just run that command. For example, if your `Makefile` is:

```
install :; forge install
```

Instead of `make install` you can run `forge install`

**Reply by Shane0xM** [reply]
Thanks @PatrickAlphaC . I used my AI buddy and couldn't find a way around the issues with `make` in my system atleast. Think the issue has to do with the MinGW installation manager, as even after reconfiguring the path in my system and GitBash `make install`  produced a repeated error.

Only solution I could find is to install Chocolatey and run command:

```shell

choco install make

**Reply by EngrPips** [reply]
I'm glad you find a way out, at least.
